PowerEditor/src/WinControls/Preference/preferenceDlg.cpp

```diff
@@ -1058,6 +1117,84 @@ intptr_t CALLBACK EditingSubDlg::run_dlgProc(UINT message, WPARAM wParam, LPARAM
+					if (wParam == IDC_RADIO_NONPRINTCHAR_CODEPOINT)
+					{
+						svp._nonPrintCharMode = svp.codepoint;
+					}
+					else // if (wParam == IDC_RADIO_NONPRINT_ABBREVIATION)
+					{
+						svp._nonPrintCharMode = svp.abbreviation;
+					}
+
+					HWND grandParent = ::GetParent(_hParent);
+					::SendMessage(grandParent, NPPM_INTERNAL_SETNONPRINTCHARS, 0, 0);
+					return TRUE;
+				}
+
+				case IDC_BUTTON_NONPRINTCHAR_NOTE_LIST:
+				{
+					NativeLangSpeaker* pNativeSpeaker = nppParam.getNativeLangSpeaker();
+					pNativeSpeaker->messageBox("NonPrintCharsListInfo",
+						_hSelf,
+						L"NOTE:\n"\
+						L"Some characters might already have some representation and are thus visible. "\
+						L"Line separator and paragraph separator are already represented by abbreviation by default.\n\n"\
+						L"Using representation will disable character effects on text.\n\n"\
+						L"Codepoint : name : abbreviation\n"\
+						L"U+00A0 : no-break space : NBSP\n"\
+						L"U+1680 : ogham space mark : OSPM\n"\
+						L"U+180E : mongolian vowel separator : MVS\n"\
+						L"U+2000 : en quad : ENQD\n"\
+						L"U+2001 : em quad : EMQD\n"\
+						L"U+2002 : en space : ENSP\n"\
+						L"U+2003 : em space : EMSP\n"\
+						L"U+2004 : three-per-em space : EMSP13\n"\
+						L"U+2005 : four-per-em space : EMSP14\n"\
+						L"U+2006 : six-per-em space : EMSP16\n"\
+						L"U+2007 : figure space : NUMSP\n"\
+						L"U+2008 : punctation space : PUNCSP\n"\
+						L"U+2009 : thin space : THINSP\n"\
+						L"U+200A : hair space : HAIRSP\n"\
+						L"U+200B : zero-width space : ZWSP\n"\
+						L"U+200C : zero-width non-joiner : ZWNJ\n"\
+						L"U+200D : zero-width joiner : ZWJ\n"\
+						L"U+200E : left-to-right mark : LRM\n"\
+						L"U+200F : right-to-left mark : RLM\n"\
+						L"U+2028 : line separator : LS\n"\
+						L"U+2029 : paragraph separator : PS\n"
+						L"U+202A : left-to-right embedding : LRE\n"\
+						L"U+202B : right-to-left embedding : RLE\n"\
+						L"U+202C : pop directional formatting : PDF\n"\
+						L"U+202D : left-to-right override : LRO\n"\
+						L"U+202E : right-to-left override : RLO\n"\
+						L"U+202F : narrow no-break space : NNBSP\n"\
+						L"U+205F : medium mathematical space : MMSP\n"\
+						L"U+2060 : word joiner : WJ\n"\
+						L"U+2066 : left-to-right isolate : LRI\n"\
+						L"U+2067 : right-to-left isolate : RLI\n"\
+						L"U+2068 : first strong isolate : FSI\n"\
+						L"U+2069 : pop directional isolate : PDI\n"\
+						L"U+206A : inhibit symmetric swapping : ISS\n"\
+						L"U+206B : activate symmetric swapping : ASS\n"\
+						L"U+206C : inhibit arabic form shaping : IAFS\n"\
+						L"U+206D : activate arabic form shaping : AAFS\n"\
+						L"U+206E : national digit shapes : NADS\n"\
+						L"U+206F : nominal digit shapes : NODS\n"\
+						L"U+3000 : ideographic space : ISP\n"\
+						L"U+FEFF : zero-width no-break space : ZWNBSP",
+						L"Representation of selected whitespace and control characters",
+						MB_OK);
+				}

```

Comment:
This dialog box is huge, and with not-well-presented information.
IMO, the display of this information is not necessarily in Notepad++, it could launch the browser to show the information in https://npp-user-manual.org/ in which we can do a pretty table easily in HTML to display such info (I did the following table in less 3 minutes in Markdown):
![image](https://user-images.githubusercontent.com/90293/216501846-c424a31b-cc2b-4851-b6e4-438e3071d05e.png)

If users don't have Internet connection, never mind - it's a Nice-to-have but not Must-to-have.
Furthermore we can save the effort for the translation and its management - less is more.